src/vms/pvm/etna-builder/spend.ts

```diff
@@ -0,0 +1,473 @@
+  output: Serializable,
+):
+  | [error: null, transferOutput: TransferOutput, locktime: bigint]
+  | [error: Error, transferOutput: null, locktime: null] => {
+  try {
+    if (isStakeableLockOut(output) && isTransferOut(output.transferOut)) {
+      return [null, output.transferOut, output.lockTime.value()];
+    } else if (isTransferOut(output)) {
+      return [null, output, 0n];
+    }
+  } catch (error) {
+    return [
+      new Error('An unexpected error occurred while unwrapping output', {
+        cause: error instanceof Error ? error : undefined,
+      }),
+      null,
+      null,
+    ];
+  }
+
+  return [new Error('Unknown output type'), null, null];
+};
+
+type SpendResult = Readonly<{
+  changeOutputs: readonly TransferableOutput[];
+  inputs: readonly TransferableInput[];
+  inputUTXOs: readonly Utxo[];
+  stakeOutputs: readonly TransferableOutput[];
+}>;
+
+type SpendProps = Readonly<{
+  /**
+   * Contains the currently accrued transaction complexity that
+   * will be used to calculate the required fees to be burned.
+   */
+  complexity: Dimensions;
+  /**
+   * Contains the amount of extra AVAX that spend can produce in
+   * the change outputs in addition to the consumed and not burned AVAX.
+   */
+  excessAVAX?: bigint;
+  /**
+   * List of Addresses that are used for selecting which UTXOs are signable.
+   */
+  fromAddresses: readonly Address[];
+  /**
+   * Optionally specifies the output owners to use for the unlocked
+   * AVAX change output if no additional AVAX was needed to be burned.
+   * If this value is `undefined`, the default change owner is used.
+   */
+  ownerOverride?: OutputOwners;
+  spendOptions: Required<SpendOptions>;
+  /**
+   * Maps `assetID` to the amount of the asset to spend without
+   * producing an output. This is typically used for fees.
+   * However, it can also be used to consume some of an asset that
+   * will be produced in separate outputs, such as ExportedOutputs.
+   *
+   * Only unlocked UTXOs are able to be burned here.
+   */
+  toBurn: Map<string, bigint>;
+  /**
+   * Maps `assetID` to the amount of the asset to spend and place info
+   * the staked outputs. First locked UTXOs are attempted to be used for
+   * these funds, and then unlocked UTXOs will be attempted to be used.
+   * There is no preferential ordering on the unlock times.
+   */
+  toStake?: Map<string, bigint>;
+  /**
+   * List of UTXOs that are available to be spent.
+   */
+  utxos: readonly Utxo[];
+}>;
+
+/**
+ * Processes the spending of assets, including burning and staking, from a list of UTXOs.
+ *
+ * @param {SpendProps} props - The properties required to execute the spend operation.
+ * @param {Context} context - The context in which the spend operation is executed.
+ *
+ * @returns {[null, SpendResult] | [Error, null]} - A tuple where the first element is either null or an error,
+ * and the second element is either the result of the spend operation or null.
+ */
+export const spend = (
```

Comment:
There is some ambiguity in all these functions regarding locktime and stakeable locktime. Can we clearly label the stakeable locktime variables as such, to avoid this ambiguity?